datafusion/sql/src/unparser/expr.rs

```diff
@@ -675,21 +675,29 @@ impl Unparser<'_> {
     }
 
     fn col_to_sql(&self, col: &Column) -> Result<ast::Expr> {
+        // Replace the column name if the dialect has an override
+        let col_name = match self.dialect.col_alias_overrides(self, &col.name)? {
+            Some(encoded_name) => encoded_name,
```

Comment:
I prefer to rename it to `rewritten_name` because the encoding way is only implemented for BigQueryDialect.
```suggestion
            Some(rewritten_name) => rewritten_name,
```